### PR TITLE
Internal/create asa vault

### PR DESCRIPTION
**Pull request by Jesulonimi21**
This branch adds support for the Asa Vault to the sdk

### CODE REVIEW

File: examples/asa-vault/redemptions.js
Comment (AwolDes) [inline examples/asa-vault/redemptions.js:15]:
```suggestion
  const account = new VaultClient({
    mnemonic,
    network: 'TestNet',
    asaVault: {decimals: 8},
  }, true);
```
I think this needs to include the second parameter to define the `asa` vault? Alternatively, we could check the existence of  `params.asaVault` in the constructor instead.

File: src/VaultClient.ts
Comment (AwolDes) [inline src/VaultClient.ts:135]:
```suggestion
    const res = await put.depositCollateral(
      this.asaVault ? convertToMicroUnits(params.amount, this.asaVault.decimals : 
      convertToMicroUnits(params.amount),
    );
```
I think this is a bit easier to read, and a bit more certain in the outcome. `undefined` is odd to pass in and could lead to unexpected behavior.

File: examples/asa-vault/redemptions.js
Comment (AwolDes) [reply examples/asa-vault/redemptions.js:15]: Looks like `VaultClient` uses this pattern but `Vault` does not - is there a reason for this?

File: src/VaultClient.ts
Comment (AwolDes) [reply src/VaultClient.ts:135]: hmm I am a bit torn on this actually given how often we do it. Would appreciate input from other team members on this

**Review by AwolDes — CHANGES_REQUESTED** [review]
Looking good! Just an inconsistency with the `constructor` method between `Vault` and `VaultClient`, and a question on how to best handle `convertToMicroUnits`

File: examples/asa-vault/redemptions.js
Comment (Jesulonimi21) [reply examples/asa-vault/redemptions.js:15]: Thanks @AwolDes 
I just thought it was unnecessary since we did not need the decimals there but have made it consistent now

File: src/VaultClient.ts
Comment (Jesulonimi21) [reply src/VaultClient.ts:136]: @AwolDes 
I passed that in because of null, a person could pass null in place of decimals and the code will error. let me know if you want me to change this

Comment (AwolDes) [reply src/VaultClient.ts:136]: I think it is ok for now, maybe something we can refactor in the future

**Review by AwolDes — APPROVED** [review]
Looks good to me, let's wait for a review from @tomxbacked or @JCastrejonE before merging

**Comment by AwolDes** [comment]
Make sure we update `VAULT_IDS` in `utils.ts` with the freshly deployed ASA vaults

**Review by tomxbacked — APPROVED** [review]

**Pull request merged**
